languages/python/concepts/lists/about.md

```diff
@@ -0,0 +1,249 @@
+
+```python
+>>> no_elements = list()
+[]
+
+#the tuple is unpacked and each element is added
+>>> multiple_elements_tuple = list(("Parrot", "Bird", 334782))
+["Parrot", "Bird", 334782]
+
+#the set is unpacked and each element is added
+>>> multiple_elements_set = list({2, 3, 5, 7, 11})
+[2,3,5,7,11]
+```
+
+Results when using a list constructor with a string or a dict may be surprising:
+
+```python
+
+#string elements (Unicode code points) are iterated through and added *individually*
+>>> multiple_elements_string = list("Timbuktu")
+['T', 'i', 'm', 'b', 'u', 'k', 't', 'u']
+
+
+>>> multiple_code_points_string = list('अभ्यास')
+['अ', 'भ', '्', 'य', 'ा', 'स']
+
+"""
+The iteration default for dictionaries is over the keys.
+To include both keys and values in a list made from a dictionary, use dict.items(),
+which will return a list of (key, value) tuples.
+"""
+source_data = {"fish": "gold", "monkey": "brown"}
+>>> multiple_elements_dict_1 = list(source_data)
+['fish', 'monkey']
+
+>>> multiple_elements_dict_2 = list(source_data.items())
+[('fish', 'gold'), ('monkey', 'brown')]
+```
+
+Because the constructor will only take _iterables_ (or nothing) as arguments, objects that are _not_ iterable will throw a syntax error. Consequently, it is much easier to create a one-item list via the literal method.
+
+```python
+
+>>> one_element = list(16)
+File "<stdin>", line 1
+    >>> one_element = list(16)
+    ^
+SyntaxError: invalid syntax
+
+>>> one_element_from_iterable = list((16,))
+[16]
+```
+
+## Accessing elements
+
+Items inside lists (_like the sequence types `string` and `tuple`_), can be accessed via 0-based index and _bracket notation_. Indexes can be from **`left`** --> **`right`** (_starting at zero_) or **`right`** --> **`left`** (_starting at -1_).
+
+| **0** | **1** | **2** | **3** | **4** | **5** |\
+ -------------------------\
+ | P | y | t | h | o | n |\
+ -------------------------\
+ |_**-6**_ |_**-5**_ |_**-4**_ |_**-3**_ |_**-2**_ |_**-1**_ | <----
+
+```python
+
+>>> breakfast_foods = ["Oatmeal", "Fruit Salad", "Eggs", "Toast"]
+
+#Oatmeal is at index 0 or index -4
+>>> first_breakfast_food = breakfast_foods[0]
+'Oatmeal'
+
+>>> first_breakfast_food = breakfast_foods[-4]
+'Oatmeal'
+```
+
+The method `.pop()` can be used to both remove and return a value at a given index:
+
+```python
+
+>>> breakfast_foods = ["Oatmeal", "Fruit Salad", "Eggs", "Toast"]
+
+#Fruit Salad is at index 1 or index -3
+>>> breakfast_foods = ["Oatmeal", "Fruit Salad", "Eggs", "Toast"]
+>>> fruit_on_the_side = breakfast_foods.pop(-3)
+'Fruit Salad'
+
+>>> print(breakfast_foods)
+['Oatmeal', 'Eggs', 'Toast']
+
+```
+
+The method `.insert()` can be used to add an element at a specific position. The index given is the element _*before which to insert*_. `list.insert(0,element)` will insert at the front of the list and `list.insert(len(list), element)` is the same as calling `list.append(element)`.
```

Comment:
_Is_ `list.insert(len(list), element)` actually the same as `list.append(element)`? Or is just _functionally_ the same?

Reply:
Good question.  I copied it out of the docs, but to be honest, I don't actually _know_.  https://docs.python.org/3/tutorial/datastructures.html